mesonbuild/backend/backends.py

```diff
@@ -701,6 +701,7 @@ def replace_outputs(self, args, private_dir, output_list):
     def get_build_by_default_targets(self):
         result = OrderedDict()
         # Get all build and custom targets that must be built by default
+        # XXX: Remove build_always in future.
         for name, t in self.build.get_targets().items():
             if t.build_by_default or t.install or t.build_always:
```

Comment:
Just remove `t.build_always` and have the `build_always` kwarg on a target set the `build_by_default` and `build_always_stale` options.

It's always better to have one codepath to do something otherwise they diverge.